File: 0189-rotate-array/0189-rotate-array.java
class Solution {
    public void rotate(int[] nums, int k) 
    {
         int arrayLength = nums.length;
        int temp [] =  new int [arrayLength];

        for (int i =0 ; i<arrayLength; i++)
        {
             temp[(i+k)%arrayLength] = nums[i];
        }

        for(int i=0; i<nums.length; i++){
            nums[i]= temp[i];
            }
        
    }
}

File: 0189-rotate-array/0189-rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        int size = nums.size();
        vector<int> temp(size);

        for (int i = 0; i < size; i++) {
            int rotate_position = (i + k) % size;
            temp[rotate_position] = nums[i];
        }

        nums = temp;
    }
};